test/src/server/helpers/collections.js

```diff
@@ -0,0 +1,337 @@
+			expect(collection).to.have.keys(
+				'id',
+				'ownerId',
+				'name',
+				'description',
+				'entityType',
+				'public',
+				'createdAt',
+				'lastModified',
+				'owner'
+			);
+		});
+		expect(orderedCollections.length).to.equal(5);
+		expect(orderedCollections).to.be.descendingBy('lastModified');
+	});
+
+	it('should return collections when passed with an offset', async () => {
+		const from = 2;
+		const size = 2;
+		const allCollections = await getOrderedPublicCollections(0, 10, null, orm);
+		const orderedCollections = await getOrderedPublicCollections(from, size, null, orm);
+		const allCollectionsSubset = allCollections.slice(from, from + size);
+		expect(orderedCollections).to.deep.equal(allCollectionsSubset);
+		expect(orderedCollections.length).to.equal(size);
+		expect(orderedCollections).to.be.descendingBy('lastModified');
+	});
+});
+
+describe('getOrderedCollectionsForEditorPage', () => {
+	// eslint-disable-next-line one-var
+	let editor, editor2;
+	before(async () => {
+		editor = await createEditor();
+		editor2 = await createEditor();
+		const collectionAttrib = {
+			description: 'description',
+			entityType: 'Author',
+			name: 'name',
+			ownerId: editor.get('id'),
+			public: true
+		};
+		// create a public collections owned by editor1
+		await new UserCollection(collectionAttrib).save(null, {method: 'insert'});
+		// create a public collections owned by editor1 and editor2 as collaborator
+		const collection = await new UserCollection(collectionAttrib).save(null, {method: 'insert'});
+		await new UserCollectionCollaborator({
+			collaboratorId: editor2.get('id'),
+			collectionId: collection.get('id')
+		}).save(null, {method: 'insert'});
+		// create a private collection owned by editor1
+		collectionAttrib.public = false;
+		await new UserCollection(collectionAttrib).save(null, {method: 'insert'});
+		// create a private collection owned by editor1 and editor2 as collaborator
+		const collection2 = await new UserCollection(collectionAttrib).save(null, {method: 'insert'});
+		await new UserCollectionCollaborator({
+			collaboratorId: editor2.get('id'),
+			collectionId: collection2.get('id')
+		}).save(null, {method: 'insert'});
+		// create one Edition collection with editor1 as owner
+		collectionAttrib.public = true;
+		collectionAttrib.entityType = 'Edition';
+		await new UserCollection(collectionAttrib).save(null, {method: 'insert'});
+	});
+	after(truncateEntities);
+
+	it('should return all collections when editor1 is logged in (fetching collections for editor1)', async () => {
+		const req = {
+			app: {
+				locals: {
+					orm
+				}
+			},
+			params: {
+				id: editor.get('id')
+			},
+			user: {
+				id: editor.get('id')
+			}
+		};
+
+		const orderedCollections = await getOrderedCollectionsForEditorPage(0, 10, null, req);
+
+		expect(orderedCollections.length).to.equal(5);
+		orderedCollections.forEach((collection) => {
+			expect(collection).to.have.keys(
```

Comment:
I don't think this is necessary in any of the `getOrderedCollectionsForEditorPage` tests you currently have.
The expect(…).to.have.keys part would instead be in a specific separate test like you did for `getOrderedPublicCollections` ("should return collections with expected keys").

Instead, I'd like you to ensure that the collections that `getOrderedCollectionsForEditorPage` returns are the ones you expect.
You can compare by the collection ids. If you map `orderedCollections` to get an array of ids, you can compare it with an array composed of the ids of the collections you're expecting to be returned.
That will make for much more robust tests, as currently you're only comparing the length of the array of returned values.